From 85132f84526add0248459ad78ea1136f4de2bd51 Mon Sep 17 00:00:00 2001
From: Brian Behlendorf <behlendorf1@llnl.gov>
Date: Thu, 3 Dec 2015 12:47:08 -0800
Subject: [PATCH 05/18] Soft disconnect when 'build_wait_timeout==0'

Insubstantiating the latent build slave without cleanly disconnecting
it can result in a ping failure if a new latent slave is immediately
instantiated.  To prevent this the slave should disconnect cleanly
before being insubstantiated.

  [Broker,2,52.25.93.92] ping finished: failure
  [Broker,2,52.25.93.92] slave ping failed; re-queueing the request

This is easily reproducible by running back to back builds using EC2.

Signed-off-by: Brian Behlendorf <behlendorf1@llnl.gov>
---
 master/buildbot/buildslave/base.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/master/buildbot/buildslave/base.py b/master/buildbot/buildslave/base.py
index 07c077b35..c28cf5043 100644
--- a/master/buildbot/buildslave/base.py
+++ b/master/buildbot/buildslave/base.py
@@ -966,7 +966,7 @@ class AbstractLatentBuildSlave(AbstractBuildSlave):
         self.building.remove(sb.builder_name)
         if not self.building:
             if self.build_wait_timeout == 0:
-                d = self.insubstantiate()
+                d = self._soft_disconnect()
                 # try starting builds for this slave after insubstantiating;
                 # this will cause the slave to re-substantiate immediately if
                 # there are pending build requests.
-- 
2.14.3

